### PR TITLE
Model specific Crystal MLP proxy and wrapper

### PR DESCRIPTION
**Pull request by sh-divya**
Proxy model specific to MLP trained on Sendek LR model. Currently accepts path to saved state_dict and standardization tensors, which are available in the data directory.

Class that wraps as well the the MLP module class are available

### CODE REVIEW

File: gflownet/proxy/crystals.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            if l % 2 == 1:
+                x = self.hidden_act(x)
+
+        return x
+
+
+if __name__ == "__main__":
+    tmp = SendekMLPWrapper(
+        "all",
+        "/Users/divya-sh/Documents/gflownet/data/crystals",
```
Comment (alexhernandezgarcia) [inline gflownet/proxy/crystals.py:108]: Ideally this should be taken from the config, specifically from the user/<username>.yaml config file.

File: gflownet/proxy/crystals.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    def __call__(self, crystals):
+        """
+        Returns a vector of size [batch_size] that are calculated
+        ionic conductivity values between 0 and 1
+        """
+
+        if self.scale is not None:
+            crystals = (crystals - self.scale["mean"]) / self.scale["std"]
+        crystals = torch.nan_to_num(crystals, nan=0.0)
```
Comment (alexhernandezgarcia) [inline gflownet/proxy/crystals.py:59]: Why would some values be nan?

File: gflownet/proxy/crystals.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        with torch.no_grad():
+            scaled_ionic_conductivity = self.oracle(crystals.to(torch.float32))
```
Comment (alexhernandezgarcia) [inline gflownet/proxy/crystals.py:61]:
```suggestion
            scaled_ionic_conductivity = self.oracle(crystals.to(device=self.device, dtype=self.float))
```
The class proxy has attributes device and float to be used for this :)

**Review by alexhernandezgarcia — CHANGES_REQUESTED** [review]
Looks great, thanks! I have suggested one (minor) change and commented about taking the path from the config instead of being hard coded. Not crucial because it won't break anything, but it would be nicer.

File: gflownet/proxy/crystals.py (hunk 2 above)
Comment (sh-divya) [reply gflownet/proxy/crystals.py:59]: the std of some features is 0, so division by zero leads to nans

Comment (alexhernandezgarcia) [reply gflownet/proxy/crystals.py:59]: Oh I see, then I would catch that directly at the normalisation line, by adding a small constant. I'll suggest a change.

Comment (alexhernandezgarcia) [reply gflownet/proxy/crystals.py:59]: Actually, if we want those values to become 0.0, then my suggestion would not work, obviously. In that case we should explicitly make those values 0.

File: config/proxy/crystals.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+_target_: gflownet.proxy.crystals.SendekMLPWrapper
```
Comment (sh-divya) [inline config/proxy/crystals.yaml:1]: Proxy and user config file for crystal runs

File: gflownet/proxy/crystals.py [hunk 5]
```diff
@@ -100,12 +100,3 @@ def forward(self, x):
                 x = self.hidden_act(x)
 
         return x
-
-
-if __name__ == "__main__":
```
Comment (sh-divya) [inline gflownet/proxy/crystals.py]: removed hardcoding of paths
